deps/wazuh_testing/wazuh_testing/process_resource_monitoring/README.md

```diff
@@ -0,0 +1,157 @@
+
+## Monitor module
+
+The `monitor` module provides a class for monitoring processes and their child processes. Monitoring is conducted concurrently, with a separate thread for each process. The following data is collected for each scan:
+
+   - **Daemon**: daemon name.
+   - **Version**: version.
+   - **Timestamp**: timestamp of the scan.
+   - **PID**: PID of the process.
+   - **CPU**(%): CPU percentage of the process. It can exceed 100% if the process uses multiple threads.
+   - **VMS**: Virtual Memory Size.
+   - **RSS**: Resident Set Size.
+   - **USS**: Unique Set Size.
+   - **PSS**: Proportional Set Size.
+   - **SWAP**: Memory of the process in the swap space.
+   - **FD**: File descriptors opened by the process.
+   - **Read_Ops**: Read operations.
+   - **Write_Ops**: Write operations.
+   - **Disk_Read**: Bytes read by the process.
+   - **Disk_Written**: Bytes written by the process.
+
+## Disk Usage Tracker module
+
+The `disk_usage_tracker` module provides a class to monitor file and directory disk usage. The following data is collected for each file or directory:
+
+- **File:** Name of the file.
+- **Timestamp:** Timestamp of the scan.
+- **Path:** Full path of the file.
+- **Size:** Size in selected units of the file.
+- **Usage:** Percentage of the space the file takes relative to the partition's size.
+- **Mod_time:** Last time the file was modified.
+- **Acc_time:** Last time the file was accessed.
+- **Creat_time:** Creation time (Windows) or metadata change time (Unix).
+
+The package also provides the script `wazuh-metrics`, designed to interact with these modules. [More information](#script)
+ 
+### Process reference
+
+#### Wazuh manager
+
+| Process | Argument |
+| ------- | -------- |
+| wazuh-agentlessd | agentlessd |
+| wazuh-analysisd | analysisd |
+| wazuh_apid.py | apid |
+| wazuh-authd | authd |
+| wazuh_clusterd.py | clusterd |
+| wazuh-csyslogd | csyslogd |
+| wazuh-db | db |
+| wazuh-dbd | dbd |
+| wazuh-execd | execd |
+| wazuh-integratord | integratord |
+| wazuh-logcollector | logcollector |
+| wazuh-maild | maild |
+| wazuh-modulesd | modulesd |
+| wazuh-monitord | monitord |
+| wazuh-remoted | remoted |
+| wazuh-syscheckd | syscheckd |
+
+> Note:  
+> `wazuh_apid.py` and `wazuh_clusterd.py` are scripts run by the Python interpreter, not processes themselves.
+
+#### Wazuh agent
+
+| Process | Argument |
+| ------- | -------- |
+| wazuh-agentd | agentd |
+| wazuh-execd | execd |
+| wazuh-logcollector | logcollector |
+| wazuh-modulesd | modulesd |
+| wazuh-syscheckd | syscheckd |
+
+
+## Directory structure
+
+```shell script
+process_resource_monitoring/
+├── pyproject.toml
+├── README.md
+└── src
+    ├── process_resource_monitoring
+    │   ├── disk_usage_tracker.py
+    │   ├── __init__.py
+    │   ├── _logger.py
+    │   └── monitor.py
+    └── wazuh_metrics.py
+```
+
+
+## Prerequisites
+
+- Wazuh component(s) 4.9.0 (or greater)
```

Comment:
Because, as mentioned in https://github.com/wazuh/wazuh-qa/pull/5747#discussion_r1771830749, at the moment of writing this this module is dependant of Wazuh, since it is only able to monitor processes with specific names that Wazuh uses for their processes.